### ISSUE TITLE
Seek not working properly (on VCF)

### ISSUE THREAD

**Issue by CastilloDel** [zaeleus/noodles#21]
The following code fails:
```rust
  let path = std::env::current_dir()
    .unwrap()
    .join("data")
    .join("vcf")
    .join("sample1-bcbio-cancer.vcf.gz");

  let mut reader = File::open(path)
    .map(bgzf::Reader::new)
    .map(vcf::Reader::new)
    .unwrap();

  reader.read_header().unwrap();
  loop {
    if reader.read_record(&mut String::new()).unwrap() == 0 {
      break;
    }
  }

  reader
    .seek(VirtualPosition::try_from((0, 9302)).unwrap())
    .unwrap();
  assert_eq!(
    reader.virtual_position(),
    VirtualPosition::try_from((0, 9302)).unwrap()
  );
```
The file I'm reading is [this one](https://github.com/umccr/htsget-rs/blob/variants/data/vcf/sample1-bcbio-cancer_uncompressed.vcf) (The compressed version can be found in the same folder).
I think it may have something to do with the EOF.

**Comment by zaeleus** [comment]
This was a bug that occurs after a seek. It's fixed in 7b1132dce0c729130ae2581c43ae188ce7b30f00.

**Comment by CastilloDel** [comment]
Thanks!
